### DISCUSSION TITLE
Need to call brew services as non-root user in script running as root

### DISCUSSION THREAD

**Discussion by gamcall** [Homebrew/discussions#2855] [Everyday usage]
Hi,

I need to run the brew services command to stop mysql in a script running as root. The problem is that command should not be run as the root user and so I am trying to use the -u and -i flags on the sudo command for:
`sudo -u <username> -i brew services stop mysql@5.7`
Unfortunately, this results in the output of:
Error: Service mysql@5.7 is not started.
This is true as far as the root user is concerned as brew services (as root) will produce:
mysql@5.7 stopped
However, when you drop down to the user where the mysql@5.7 service was started, it is still running.
What am I doing wrong here? Can anyone help? 
Thanks,
GAM

**Comment by SMillerDev** [comment]
`sudo -i` means interactive so you want to omit that. Either way, "how do I impersonate a user as root" isn't really a Homebrew question. I'd suggest checking Stackoverflow for this.

**Reply by gamcall** [reply]
Classic. Apple Discussions & Ask Different referred me here.
From the man page of sudo on my Mac:

`-i, --login Run the shell specified by the target user's password database entry as a login shell`

I found I needed to do that or I wasn’t getting any environment or context for the brew services call.

Regards,

GAM
